File: mySqrt.py
# -*- coding: utf-8 -*-
"""
Created on Mon Oct 19 14:54:28 2015

@author: weizhi
"""

class Solution(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x==0:
            return 0
        left = 1
        right = x
      #  right = x/2 + 1
        while left<=right:
            mid = left + (right-left)/2
           # print mid
            if mid**2==x:
                return mid
            elif mid**2>x:
                right = mid-1
            else:
                left = mid+1
        return right
        
        
obj = Solution()
print obj.mySqrt(5)

File: leetcodeStudy_Java_Solution/src/leetcodeStudy/mySqrt.java
package leetcodeStudy;

public class mySqrt {
	public int mySqrt(int x){
		if(x<=0)
			return 0;
		int l = 1;
		int r = x;
		while(l<r){
			int mid = r + (l-r)/2;
			if (mid*mid==x)
				return mid;
			else if (mid*mid<x)
				l = mid+1;
			else
				r = mid-1;
				
		}
		return l;
	}

}


